spec/services/metrics/merge_time/per_user_project_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'rails_helper'
+
+RSpec.describe Metrics::MergeTime::PerUserProject do
+  describe '.call' do
+    let(:user_project) { create(:users_project) }
+    let(:pull_request) do
+      create(:pull_request, state: :open,
+                            project_id: user_project.project_id,
+                            owner_id: user_project.user_id)
+    end
+
+    context 'when processing a collection containing no pull request events' do
+      it 'does not create a metric' do
+        expect { described_class.call }.not_to change { Metric.count }
+      end
+    end
+
+    context 'Generating metrics values' do
+      context 'when merge ocurred in a 30 minutes interval' do
+        let!(:pull_request) do
+          create(:pull_request,
+                 opened_at: Time.zone.now,
+                 merged_at: Time.zone.now + 30.minutes,
+                 project_id: user_project.project_id,
+                 owner_id: user_project.user_id)
+        end
+
+        it 'generates a metric with value expressed as decimal equal to 30 minutes' do
+          described_class.call
+          expect(Metric.last.value.seconds).to eq(30.minutes)
+        end
+
+        it 'generates only that metric' do
+          expect { described_class.call }.to change { Metric.count }.from(0).to(1)
+        end
+      end
+
+      context 'when calculating the merge time value' do
+        context 'and a user made more than one pull request' do
+          let!(:pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 30.minutes,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          let!(:second_pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 45.minutes,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          let!(:third_pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 2.hours,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          it 'creates just one metric' do
```

Comment:
Can we add another test to make sure the average is calculated correctly in this case?